allennlp/data/fields/text_field.py

```diff
@@ -72,15 +72,14 @@ def get_padding_lengths(self) -> Dict[str, int]:
                 # _empty_ TextField, but if this is the case, token_lengths will be an empty
                 # list, so we add the default empty padding dictionary to the list instead.
                 token_lengths = [{}]
-            # Iterate over the keys in the first element of the list.
-            # This is fine as for a given indexer, all tokens will return the same keys,
-            # so we can just use the first one.
+            # Iterate over the keys and find the maximum token length.
+            # It's fine to iterate over the keys of the first token since all tokens have the same keys.
             for key in token_lengths[0].keys():
                 indexer_lengths[key] = max(x[key] if key in x else 0 for x in token_lengths)
             lengths.append(indexer_lengths)
         any_indexed_token_key = list(self._indexed_tokens.keys())[0]
         padding_lengths = {'num_tokens': len(self._indexed_tokens[any_indexed_token_key])}
-        # Get all the keys which have been used for padding.
+        # Get all keys which have been used for padding for each indexer and take the max if there are duplicates.
```

Comment:
Done.  I'm glad you think the max only resolves conflicts in unlikely cases.  I wonder if we should have asserts rather than code to handle potentially non-existent outliers.

Reply:
In this case, I think the max is the right thing to do, not an assert.  An assert would stop you in some rare but conceivable use cases.  E.g., @DeNeutoy recently had a case where he wanted to have two identical `TokenIndexers`, so he could embed the same array in two different ways in his model without too much work.  That would give you duplicate keys, and an assert would have prevented him from doing what he wanted, but a max does the right thing in that case.

Reply:
Cool, I didn't realize we had actual examples of when we want this behavior.